src/common/servers/serverManager.ts

```diff
@@ -0,0 +1,442 @@
+    }
+
+    getAllServers = () => {
+        return [...this.servers.values()];
+    }
+
+    hasServers = () => {
+        return Boolean(this.servers.size);
+    }
+
+    getRemoteInfo = (serverId: string) => {
+        return this.remoteInfo.get(serverId);
+    }
+
+    updateRemoteInfos = (remoteInfos: Map<string, RemoteInfo>) => {
+        let hasUpdates = false;
+        remoteInfos.forEach((remoteInfo, serverId) => {
+            this.remoteInfo.set(serverId, remoteInfo);
+            hasUpdates = this.updateServerURL(serverId) || hasUpdates;
+            hasUpdates = this.openExtraTabs(serverId) || hasUpdates;
+        });
+
+        if (hasUpdates) {
+            this.persistServers();
+        }
+    }
+
+    lookupTabByURL = (inputURL: URL | string, ignoreScheme = false) => {
+        log.silly('lookupTabByURL', `${inputURL}`, ignoreScheme);
+
+        const parsedURL = urlUtils.parseURL(inputURL);
+        if (!parsedURL) {
+            return undefined;
+        }
+        const server = this.getAllServers().find((server) => {
+            return equalUrlsIgnoringSubpath(parsedURL, server.url, ignoreScheme) && parsedURL.pathname.match(new RegExp(`^${server.url.pathname}(.+)?(/(.+))?$`));
+        });
+        if (!server) {
+            return undefined;
+        }
+        const tabs = this.getOrderedTabsForServer(server.id);
+
+        let selectedTab = tabs.find((tab) => tab && tab.name === TAB_MESSAGING);
+        tabs.
+            filter((tab) => tab && tab.name !== TAB_MESSAGING).
+            forEach((tab) => {
+                if (parsedURL.pathname.match(new RegExp(`^${tab.url.pathname}(/(.+))?`))) {
+                    selectedTab = tab;
+                }
+            });
+        return selectedTab;
+    }
+
+    updateServerOrder = (serverOrder: string[]) => {
+        log.debug('updateServerOrder', serverOrder);
+
+        this.serverOrder = serverOrder;
+        this.persistServers();
+    }
+
+    updateTabOrder = (serverId: string, tabOrder: string[]) => {
+        log.debug('ServerManager.updateTabOrder', serverId, tabOrder);
+
+        this.tabOrder.set(serverId, tabOrder);
+        this.persistServers();
+    }
+
+    addServer = (server: Team) => {
+        const newServer = new MattermostServer(server, false);
+
+        if (this.servers.has(newServer.id)) {
+            throw new Error('ID Collision detected. Cannot add server.');
+        }
+        this.servers.set(newServer.id, newServer);
+
+        this.serverOrder.push(newServer.id);
+        const tabOrder: string[] = [];
+        getDefaultTabs().forEach((tab) => {
+            const newTab = this.getTabView(newServer, tab.name, tab.isOpen);
+            this.tabs.set(newTab.id, newTab);
+            tabOrder.push(newTab.id);
+        });
+        this.tabOrder.set(newServer.id, tabOrder);
+
+        // Emit this event whenever we update a server URL to ensure remote info is fetched
+        this.emit(SERVERS_URL_MODIFIED, [newServer.id]);
```

Comment:
I suppose it could be, but users are already used to doing this. Doesn't make too much sense to remove the functionality to me, and I can see where people might want to change the URL (if something updates) so we can leave it.